src/ripple/rpc/handlers/Subscribe.cpp

```diff
@@ -71,25 +71,31 @@ Json::Value doSubscribe (RPC::Context& context)
 
         if (!ispSub)
         {
-            WriteLog (lsDEBUG, RPCHandler)
+            JLOG (context.j.debug)
                 << "doSubscribe: building: " << strUrl;
 
-            RPCSub::pointer rspSub = RPCSub::New (context.app.getOPs (),
+            auto rspSub = make_RPCSub (context.app.getOPs (),
                 context.app.getIOService (), context.app.getJobQueue (),
-                    strUrl, strUsername, strPassword);
+                    strUrl, strUsername, strPassword, context.app.logs ());
             ispSub  = context.netOps.addRpcSub (
                 strUrl, std::dynamic_pointer_cast<InfoSub> (rspSub));
         }
         else
         {
-            WriteLog (lsTRACE, RPCHandler)
+            JLOG (context.j.trace)
                 << "doSubscribe: reusing: " << strUrl;
 
-            if (context.params.isMember (jss::username))
-                dynamic_cast<RPCSub*> (&*ispSub)->setUsername (strUsername);
-
-            if (context.params.isMember (jss::password))
-                dynamic_cast<RPCSub*> (&*ispSub)->setPassword (strPassword);
+            if (auto rpcSub = std::dynamic_pointer_cast<RPCSub> (ispSub))
```

Comment:
Since we don't need a `shared_ptr` for `rpcSub` I'd prefer `dynamic_cast` over `dynamic_pointer_cast` here. Do we always expect `ispSub` to be of type `RPCSub` (the old code did). Should we `assert` if it isn't?

(This is not important enough to hold up the PR, it just caught my eye).